das/worker.go

```diff
@@ -0,0 +1,97 @@
+package das
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/header"
+
+	"go.uber.org/multierr"
+)
+
+type worker struct {
+	lock  sync.Mutex
+	state workerState
+}
+
+// workerState contains important information about the state of a
+// current sampling routine.
+type workerState struct {
+	job
+
+	Curr   uint64 `json:"current"`
+	Err    error  `json:"error,omitempty"`
+	failed []uint64
+}
+
+type job struct {
+	id   int
+	From uint64 `json:"from"`
+	To   uint64 `json:"to"`
+}
+
+func (w *worker) run(
+	ctx context.Context,
+	getter header.Getter,
+	sample sampleFn,
+	resultCh chan<- result) {
+	jobStart := time.Now()
+	log.Debugw("start sampling worker", "from", w.state.From, "to", w.state.To)
+
+	for curr := w.state.From; curr <= w.state.To; curr++ {
+		// TODO: get headers in batches
+		h, err := getter.GetByHeight(ctx, curr)
```

Comment:
What is stopping us from getting ranges from getter now? If we want to keep it in another PR, let's make an issue.

Reply:
new PR please

Reply:
There is an issue for that:
https://github.com/celestiaorg/celestia-node/issues/1015
Skipped it for now, because I need some time to dig deeper into `GetRangeByHeight` blocking mechanics.